inference/core/interfaces/camera/video_source.py

```diff
@@ -0,0 +1,1015 @@
+import os
+import time
+from dataclasses import dataclass
+from datetime import datetime
+from enum import Enum
+from queue import Empty, Queue
+from threading import Event, Lock, Thread
+from typing import Any, Callable, List, Optional, Protocol, Union
+
+import cv2
+import supervision as sv
+
+from inference.core import logger
+from inference.core.interfaces.camera.entities import (
+    StatusUpdate,
+    UpdateSeverity,
+    VideoFrame,
+)
+from inference.core.interfaces.camera.exceptions import (
+    EndOfStreamError,
+    SourceConnectionError,
+    StreamOperationNotAllowedError,
+)
+
+DEFAULT_BUFFER_SIZE = int(os.getenv("VIDEO_SOURCE_BUFFER_SIZE", "64"))
```

Comment:
Should these configs live in `inference/core/env.py`?

Reply:
👍